### ISSUE TITLE
False positive in `unnecessary_unwrap`

### ISSUE THREAD

**Issue by JohnTitor** [rust-lang/rust-clippy#5174]
@flip1995 pointed out this FP in `unnecessary_unwrap`:

```rust
macro_rules! m {
    ($a:expr, $b:expr) => {
         if $a {
              $b;
         }
    }
}

let x: Option<u32> = Some(42);
m!(x.is_some(), x.unwrap());
```

([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=0951a843a77ca548f7f23e69740c8b82))

We should also cover this case, it's a follow-up of #5132.

_Originally posted by @flip1995 in https://github.com/rust-lang/rust-clippy/pull/5132#issuecomment-582124786_
